src/main/java/Deadline.java

```diff
@@ -0,0 +1,31 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    LocalDate deadline;
+    public Deadline(String description, String deadline) {
+        super(description);
+        this.deadline = LocalDate.parse(deadline);
+    }
+
+    public Deadline(String description, String deadline, boolean isDone) {
+        super(description, isDone);
+        this.deadline = LocalDate.parse(deadline);
+    }
+
+    /**
+     * Converts the details of the file into the format represented in the storage file.
+     *
+     * @return A string in the appropriate format, E | Completion Status (0 or 1) | Description | Deadline
+     */
+    @Override
+    public String convertToFileString() {
+        return "D | " + super.convertToFileString() + String.format(" | %s\n", deadline);
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString()
+                + String.format(" (by: %s)", deadline.format(DateTimeFormatter.ofPattern("MMM d yyyy")));
```

Comment:
Good job placing line breaks to improve readability! Maybe have the same indentation for "deadline.format..." as well?